src/main/java/org/sonarlint/intellij/monitoring/MonitoringService.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2025 SonarSource Sàrl
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.monitoring
+
+import com.intellij.openapi.application.ApplicationInfo
+import io.sentry.Sentry
+import io.sentry.SentryOptions
+import org.apache.commons.lang3.SystemUtils
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import org.sonarlint.intellij.common.util.SonarLintUtils.getService
+import org.sonarlint.intellij.telemetry.SonarLintTelemetry
+
+object MonitoringService {
+
+    private const val CLIENT_DSN = "https://68144d3abca48b0ea9e3ca1a6a24bf74@o1316750.ingest.us.sentry.io/4510509893353472"
+
+    private var active = false
+
+    @JvmStatic
+    fun init() {
+        if (!active) {
+            active = true
+            getService(SonarLintTelemetry::class.java).enabled().thenAccept { isEnabled ->
+                if (isEnabled) {
+                    initializeSentry()
+                } else {
+                    active = false
+                }
+            }
+        }
+    }
+
+    @JvmStatic
+    fun close() {
+        if (active) {
+            active = false
+            Sentry.close()
+        }
+    }
+
+    @JvmStatic
+    fun reinit() {
+        if (active) {
+            Sentry.close()
+        }
```

Comment:
Why restart it? If it's already active, couldn't we do nothing? Do we need this reinit at all?